edb/server/protocol/binary_v0.pyx

```diff
@@ -1031,6 +1034,9 @@ cdef class EdgeConnectionBackwardsCompatible(EdgeConnection):
 
             compiled = self._last_anon_compiled
 
+        if self.server.is_readonly():
+            allow_capabilities &= ~enums.Capability.WRITE
```

Comment:
We should pass a `detail` text to `make_error` below saying "the instance is in read-only mode" so users have a better explanation why the query is being rejected.